main/src/ca/uwaterloo/flix/language/ast/MonoType.scala

```diff
@@ -56,6 +56,8 @@ object MonoType {
 
   case object Region extends MonoType
 
+  case class ErasedType(kind: Kind) extends MonoType
```

Comment:
I would much prefer if this was called AnyType or UndeterminedType or really anything that does not use the word erased.